src/site/content/en/blog/sms-receiver-announce/index.md

```diff
@@ -0,0 +1,401 @@
+user.
+
+<video autoplay loop muted playsinline>
+  <source src="https://storage.googleapis.com/web-dev-assets/sms-receiver-announce/demo.mp4" type="video/mp4">
+</video>
+
+Notice that as the text message arrives, the bottom sheet appears and pressing
+**Verify** allows the OTP to be autofilled in the form or even submitted
+automatically, signing in the user. The whole process is diagrammed in the image
+below.
+
+<figure class="w-figure w-figure--center">
+  <img src="./diagram.png" width="486" height="499" />
+  <figcaption class="w-figcaption w-figcaption--fullbleed">
+    SMS Receiver API diagram
+  </figcaption>
+</figure
+
+Try [the
+](https://sms-receiver-demo.glitch.me)[demo](https://sms-receiver-demo.glitch.me)
+yourself. It doesn't ask for your phone number and send SMS to your device, but
+you can send one from another device by copying the text displayed on the demo
+because it doesn't matter who the sender is in the SMS Receiver API.
+
+1. Prepare Google Chrome on Android 78 or later with "Experimental Web Platform
+   Feature" flag turned on at
+   `chrome://flags/\#enable-experimental-web-platform-features`.
+1. Go to
+   [https://sms-receiver-demo.glitch.me](https://sms-receiver-demo.glitch.me).
+1. Select your browser channel from the provided list.
+1. Press **Copy** to copy the text message and send it to another phone.
+1. Press **Verify**.
+1. From the other phone, send yourself the copied text message via SMS.
+
+Did you receive the SMS and see the prompt to enter the code to the input area?
+That is how the SMS Receiver API works for end users.
+
+{% Aside 'warning' %}
+If you are using Chrome in Android's Work Profile feature, it will not be
+able to see SMS messages because applications in the Work Profile cannot see
+information in Personal Profile applications, including the Messages
+application.
+{% endAside %}
+
+## Using the SMS Receiver API
+
+Using the SMS Receiver API consists of two parts: JavaScript code in your appon
+your website and the formatted message text sent via SMS.
+
+{% Aside %}
+Note: The SMS Receiver API requires a secure origin (HTTPS).
+{% endAside %}
+
+### Feature detection
+
+Feature detection is much the same as for many other APIs:
+
+```js
+if ('sms' in navigator) {
+  ...
+}
+```
+
+### Process the OTP
+
+The SMS Receiver API itself is simple enough:
+
+```js
+const sms = await navigator.sms.receive();
+```
+
+Once a user taps **Verify**, displayed in the bottom sheet, the promise
+containing the text message will resolve. You can use a regular expression to
+distill the OTP and verify the user. For example, if a text message contains a
+six digit verification code following `otp=`, the code would look like this:
+
+```js
+const code = sms.content.replace(/[\s\S]\*otp=([0-9a-zA-Z]{6})[\s\S]\*$/m, '$1');
+```
+
+You can now submit the code to the server to verify it.
+
+### Designing the SMS message
+
+The API itself should have looked simple enough, but a critical part is to
```

Comment:
```suggestion
```
This sentence and the next one are a bit redundant. I think we can delete this first sentence.